src/nvim/marktree.c

```diff
@@ -0,0 +1,1157 @@
+#include <assert.h>
```

Comment:
would be good to have a module description here similar to `multiqueue.c`, also mentioning common operations like `marktree_itr/del_get`.

Anyways, really cool stuff!